common/src/main/java/com/wynntils/features/user/inventory/InventoryEmeraldCountFeature.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user.inventory;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.config.Config;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.features.properties.FeatureCategory;
+import com.wynntils.core.features.properties.FeatureInfo;
+import com.wynntils.mc.event.ContainerRenderEvent;
+import com.wynntils.mc.objects.CommonColors;
+import com.wynntils.mc.render.FontRenderer;
+import com.wynntils.mc.render.HorizontalAlignment;
+import com.wynntils.mc.render.RenderUtils;
+import com.wynntils.mc.render.Texture;
+import com.wynntils.mc.render.VerticalAlignment;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.utils.StringUtils;
+import com.wynntils.wynn.objects.EmeraldSymbols;
+import com.wynntils.wynn.utils.ContainerUtils;
+import com.wynntils.wynn.utils.InventoryUtils;
+import java.util.Arrays;
+import java.util.Objects;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen;
+import net.minecraft.client.gui.screens.inventory.InventoryScreen;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import org.lwjgl.glfw.GLFW;
+
+@FeatureInfo(category = FeatureCategory.INVENTORY)
+public class InventoryEmeraldCountFeature extends UserFeature {
+    private static final int TEXTURE_SIZE = 28;
+
+    @Config
+    public EmeraldCountType emeraldCountType = EmeraldCountType.Texture;
+
+    @Config
+    public boolean showInventoryEmeraldCount = true;
+
+    @Config
+    public boolean showContainerEmeraldCount = true;
+
+    @SubscribeEvent
+    public void onContainerRender(ContainerRenderEvent event) {
+        int emeralds =
+                ContainerUtils.getEmeraldCountInContainer(event.getScreen().getMenu());
```

Comment:
Could this be cached and updated as necessary?

Reply:
It might be hard to keep it updated consistently.